vm/systemSmartContracts/auction.go

```diff
@@ -517,17 +517,27 @@ func (s *stakingAuctionSC) getVerifiedBLSKeysFromArgs(txPubKey []byte, args [][]
 	blsKeys := make([][]byte, 0)
 	maxNodesToRun := big.NewInt(0).SetBytes(args[0]).Uint64()
 
+	invalidBlsKeys := make([]string, 0)
 	for i := uint64(1); i < maxNodesToRun*2+1; i += 2 {
 		blsKey := args[i]
 		signedMessage := args[i+1]
 		err := s.sigVerifier.Verify(txPubKey, signedMessage, blsKey)
 		if err != nil {
-			s.eei.AddReturnMessage("invalid BLS key: " + err.Error())
+			invalidBlsKeys = append(invalidBlsKeys, hex.EncodeToString(txPubKey))
 			continue
 		}
 
 		blsKeys = append(blsKeys, blsKey)
 	}
+	if len(invalidBlsKeys) != 0 {
+		returnMessage := "invalid BLS keys: "
```

Comment:
`returnMessage := "invalid BLS keys: " + strings.Join(invalidBlsKeys, ", ")`

No need for the last removal (L538)

Reply:
done.